examples/zigzag/main.go

```diff
@@ -0,0 +1,272 @@
+			}
+		}
+	}
+	return false
+}
+
+func (g *Game) wallCollision() bool {
+	if g.scalesLoc[0].X > screenWidth/2 ||
+		g.scalesLoc[0].X < -(screenWidth/2) ||
+		g.scalesLoc[0].Y > screenHeight/2 ||
+		g.scalesLoc[0].Y < -(screenHeight/2) {
+		return true
+	}
+	return false
+}
+
+func (g *Game) updateTimer() {
+	if g.timer > 0xFFFFFFFFFFFFFFFE {
+		g.timer = 0
+	}
+	g.timer++
+}
+
+func (g *Game) moveSnakeTimer() bool {
+	if g.timer%uint64(g.speed) == 0 {
+		return true
+	}
+	return false
+}
+
+func (g *Game) reset() {
+	g.AppleX = 30
+	g.AppleY = 30
+	g.speed = 5 // means 83.33 ms
+	l := len(g.scalesLoc)
+	for i := int64(1); i < int64(l); i++ {
+		delete(g.scalesLoc, i)
+	}
+	g.scalesLoc[0].X = 0
+	g.scalesLoc[0].Y = 0
+	g.Score = 0
+	g.Level = 1
+	g.moveDirection = 0
+}
+
+// Update ...
+func (g *Game) Update(screen *ebiten.Image) error {
+
+	if inpututil.IsKeyJustPressed(ebiten.KeyLeft) || inpututil.IsKeyJustPressed(ebiten.KeyA) {
+		if g.moveDirection != 2 {
+			g.moveDirection = 1
+		}
+	} else if inpututil.IsKeyJustPressed(ebiten.KeyRight) || inpututil.IsKeyJustPressed(ebiten.KeyD) {
+		if g.moveDirection != 1 {
+			g.moveDirection = 2
+		}
+	} else if inpututil.IsKeyJustPressed(ebiten.KeyDown) || inpututil.IsKeyJustPressed(ebiten.KeyS) {
+		if g.moveDirection != 4 {
+			g.moveDirection = 3
+		}
+	} else if inpututil.IsKeyJustPressed(ebiten.KeyUp) || inpututil.IsKeyJustPressed(ebiten.KeyW) {
+		if g.moveDirection != 3 {
+			g.moveDirection = 4
+		}
+	} else if inpututil.IsKeyJustPressed(ebiten.KeyEscape) {
+		g.reset()
+	}
+
+	if g.moveSnakeTimer() {
+		if g.selfCollision() {
+			g.reset()
+		}
+		if g.wallCollision() {
+			g.reset()
+		}
+		g.nearapple = g.nearApple()
+
+		if g.collision() {
+			g.AppleX = rand.Int63n(screenWidth-20) + -(screenWidth / 2)
+			g.AppleY = rand.Int63n(screenHeight-20) + -(screenHeight / 2)
+			g.scalesLoc[int64(len(g.scalesLoc))] = &Scale{
+				X: g.scalesLoc[int64(len(g.scalesLoc)-1)].X,
+				Y: g.scalesLoc[int64(len(g.scalesLoc)-1)].Y,
+			}
+			if len(g.scalesLoc) > 7 && len(g.scalesLoc) < 15 {
+				g.Level = 2
+				g.speed = 4
+			} else if len(g.scalesLoc) > 14 && len(g.scalesLoc) < 21 {
+				g.Level = 3
+				g.speed = 3
+			} else if len(g.scalesLoc) > 20 && len(g.scalesLoc) < 30 {
+				g.Level = 4
+				g.speed = 2
+			} else if len(g.scalesLoc) > 29 {
+				g.Level = 5
+				g.speed = 1
+			} else {
+				g.Level = 1
+			}
+			g.Score++
+			if g.bestScore < g.Score {
+				g.bestScore = g.Score
+			}
+		}
+
+		for i := int64(len(g.scalesLoc)) - 1; i > 0; i-- {
+			g.scalesLoc[i].X = g.scalesLoc[i-1].X
+			g.scalesLoc[i].Y = g.scalesLoc[i-1].Y
+		}
+		switch g.moveDirection {
+		case 1:
+			g.scalesLoc[0].X = g.scalesLoc[0].X - 10
+		case 2:
+			g.scalesLoc[0].X += 10
+		case 3:
+			g.scalesLoc[0].Y += 10
+		case 4:
+			g.scalesLoc[0].Y -= 10
+		}
+	}
+
+	g.updateTimer()
+
+	return nil
+}
+
+// Draw ...
+func (g *Game) Draw(screen *ebiten.Image) {
+	screen.Fill(color.RGBA{0x00, 0x00, 0x00, 0xFF})
```

Comment:
As the screen is always cleared automatically, we don't need to call this.

Reply:
done